### ISSUE TITLE
`--auto-gen-config --auto-gen-only-exclude` should generate a list of LineLength offending files

### ISSUE THREAD

**Issue by brkn** [rubocop/rubocop#12819]
## Expected behavior

Running `rubocop --auto-gen-config --auto-gen-only-exclude` should create a list of  LineLength offending files. Example before version 1.62.1

```yaml
# Offense count: 231
# This cop supports safe autocorrection (--autocorrect).
# Configuration parameters: Max, AllowHeredoc, AllowURI, URISchemes, IgnoreCopDirectives, AllowedPatterns.
# URISchemes: http, https
Layout/LineLength:
  Exclude:
    - 'app/models/book.rb'
    - 'app/models/author.rb'
```

## Actual behavior

The command outputs this instead.
```
Phase 1 of 2: run Layout/LineLength cop (skipped because only excludes will be generated due to `--auto-gen-only-exclude` flag)
Phase 2 of 2: run all cops
```

And then todo file contains this block instead

```yaml
# This cop supports safe autocorrection (--autocorrect).
# Configuration parameters: AllowHeredoc, AllowURI, URISchemes, IgnoreCopDirectives, AllowedPatterns.
# URISchemes: http, https
Layout/LineLength:
  Max: 842
```

This behaviour changed at https://github.com/rubocop/rubocop/pull/12730

## Steps to reproduce the problem

1. Have files violating the `Layout/LineLength` rule. 
1. run `rubocop --auto-gen-config --auto-gen-only-exclude`

## RuboCop version

```sh
❯ bundle exec rubocop -V 
1.62.1 (using Parser 3.3.0.5, rubocop-ast 1.31.2, running on ruby 2.7.8) [arm64-darwin22]
  - rubocop-capybara 2.20.0
  - rubocop-factory_bot 2.25.1
  - rubocop-rails 2.24.1
  - rubocop-rspec 2.29.1
```

**Comment by brkn** [comment]
Nevermind, this happens because I switched from `--exclude-limit 9999999` to `--no-exclude-limit`
